src/core/mutationObserver.ts

```diff
@@ -137,7 +137,7 @@ export class MutationObserver<
       isIdle: state.status === 'idle',
       mutate: this.mutate,
       reset: this.reset,
-    }
+    } as MutationObserverResult<TData, TError, TVariables, TContext>
```

Comment:
could we maybe do the same thing here as we do in the queryObserver, which is: declare the type as  `MutationObserverBaseResult` and then only cast it after that, see:

https://github.com/tannerlinsley/react-query/blob/f9b23fcae9c5d45e3985df4519dd8f78a9fa364e/src/core/queryObserver.ts#L544-L569

I think this would give a bit more saftey because things we set here would need to confirm to the BaseResult type, which I think they don't need with a pure type assertion